server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationClusterSettingIT.java

```diff
@@ -123,4 +133,160 @@ public void testIndexReplicationSettingOverridesDocRepClusterSetting() throws Ex
+        final String targetIndexName = resizeActionTuple.v1().name().toLowerCase(Locale.ROOT) + "-target";
+        final int documentCount = scaledRandomIntBetween(1, 10);
+
+        logger.info("--> Performing resize action {} with shard count {}", resizeActionTuple.v1(), resizeActionTuple.v2());
+
+        // Create an index using current cluster level settings
+        BiConsumer<List<ReplicationType>, List<String>> consumer = (replicationList, dataNodesList) -> {
+            Settings indexSettings = Settings.builder()
+                .put(indexSettings())
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, initialShardCount)
+                .put(SETTING_REPLICATION_TYPE, replicationList.get(0))
+                .build();
+            createIndex(INDEX_NAME, indexSettings);
+
+            for (int i = 0; i < documentCount; i++) {
+                client().prepareIndex(INDEX_NAME).setId(String.valueOf(i)).setSource("foo", "bar").get();
+            }
+
+            // Block writes
+            client().admin()
+                .indices()
+                .prepareUpdateSettings(INDEX_NAME)
+                .setSettings(Settings.builder().put("index.blocks.write", true))
+                .get();
+            ensureGreen();
+
+            try {
+                for (String node : dataNodesList) {
+                    assertBusy(() -> {
+                        assertHitCount(
+                            client(node).prepareSearch(INDEX_NAME).setSize(100).setQuery(new TermsQueryBuilder("foo", "bar")).get(),
+                            documentCount
+                        );
+                    }, 30, TimeUnit.SECONDS);
+                }
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+
+            // Test create index fails
+            client().admin()
+                .indices()
+                .prepareResizeIndex(INDEX_NAME, targetIndexName)
+                .setResizeType(resizeActionTuple.v1())
+                .setSettings(
+                    Settings.builder()
+                        .put("index.number_of_replicas", 0)
+                        .put("index.number_of_shards", resizeActionTuple.v2())
+                        .putNull("index.blocks.write")
+                        .put(SETTING_REPLICATION_TYPE, replicationList.get(1))
+                        .build()
+                )
+                .get();
+        };
+        executeTest(true, consumer, targetIndexName, documentCount);
+    }
+
+    // Creates a cluster with mis-matching cluster level and index level replication strategies and validates that index
+    // creation fails when cluster level setting `cluster.force.index.replication.type` is set to true and creation goes
+    // through when it is false.
+    private void executeTest(
+        boolean restrictIndexLevelReplicationTypeSetting,
```

Comment:
I think every invocation of this method passes `true` for this boolean, which suggests you don't need this parameter at all. I think that means you don't need the `else` branch on line 266 below and maybe you don't need `documentCount` either?

Reply:
Thanks @andrross for catching this. Yes, this flag is not needed anymore as I removed the all tests where `restrictIndexLevelReplicationTypeSetting` was sent as `false` i.e. allow index creation to go through. Also, since current tests block index creations, there is no need of indexing. Removing all this redundant code.